lib/src/user.dart

```diff
@@ -118,15 +133,29 @@ enum UserState {
 
 /// The user identity associated with a [User]
 class UserIdentity {
-  /// The unique identifier for this [UserIdentity]
+  /// The unique identifier for this [UserIdentity].
   final String id;
 
-  /// The authentication provider defining this identity
+  /// The [AuthProviderType] defining this identity.
   final AuthProviderType provider;
 
   const UserIdentity._(this.id, this.provider);
 }
 
+class UserProfile {
```

Comment:
The user profile may hold other properties - not only these. My suggestion would be to change these to be getter-only properties and to expose an indexer on the profile for accessing other properties. Something like:

```dart
class UserProfile {
    final Map<String, dynamic> _data;

    String? get name => this['name'];

    // Rest of named properties

    dynamic operator[](String property) => _data[property];

    UserProfile._(this._data);
}
```

Reply:
I just modeled it after .NET. There are not such methods. But I think it wont hurt adding them. Will do